chain-impl-mockchain/benches/tally.rs

```diff
@@ -0,0 +1,168 @@
+    .collect()
+}
+
+fn tally_benchmark(c: &mut Criterion) {
+    const MEMBERS_NO: usize = 3;
+    const THRESHOLD: usize = 2;
+    let favorable = Choice::new(1);
+
+    let mut wallets: Vec<&mut WalletTemplateBuilder> = Vec::new();
+
+    let mut alice_wallet_builder = wallet(ALICE);
+    alice_wallet_builder
+        .with(1_000)
+        .owns(STAKE_POOL)
+        .committee_member();
+    wallets.push(&mut alice_wallet_builder);
+
+    let voters_aliases = voters_aliases(VOTERS_COUNT);
+    let mut wallet_builders: Vec<WalletTemplateBuilder> =
+        voters_aliases.iter().map(|alias| wallet(alias)).collect();
+
+    for wallet_builder in wallet_builders.iter_mut() {
+        wallet_builder.with(10);
+        wallets.push(wallet_builder);
+    }
+
+    let mut rng = ChaCha20Rng::from_seed([0u8; 32]);
+    let members = CommitteeMembersManager::new(&mut rng, THRESHOLD, MEMBERS_NO);
+
+    let committee_keys = members
+        .members()
+        .iter()
+        .map(|committee_member| committee_member.public_key())
+        .collect::<Vec<_>>();
+
+    let (mut ledger, controller) = prepare_scenario()
+        .with_config(
+            ConfigBuilder::new(0)
+                .with_fee(LinearFee::new(0, 0, 0))
+                .with_rewards(Value(1000)),
+        )
+        .with_initials(wallets)
+        .with_vote_plans(vec![vote_plan(VOTE_PLAN)
+            .owner(ALICE)
+            .consecutive_epoch_dates()
+            .payload_type(PayloadType::Private)
+            .committee_keys(committee_keys)
+            .with_proposal(
+                proposal(VoteTestGen::external_proposal_id())
+                    .options(3)
+                    .action_transfer_to_rewards(100),
+            )])
+        .build()
+        .unwrap();
+
+    let mut alice = controller.wallet(ALICE).unwrap();
+
+    let vote_plan_def = controller.vote_plan(VOTE_PLAN).unwrap();
+    let vote_plan: VotePlan = vote_plan_def.clone().into();
+    let proposal = vote_plan_def.proposal(0);
+
+    for alias in voters_aliases {
+        let mut private_voter = controller.wallet(&alias).unwrap();
+
+        controller
+            .cast_vote_private(
+                &private_voter,
+                &vote_plan_def,
+                &proposal.id(),
+                favorable,
+                &mut ledger,
+                &mut rng,
+            )
+            .unwrap();
+        private_voter.confirm_transaction();
+    }
+
+    ledger.fast_forward_to(BlockDate {
+        epoch: 1,
+        slot_id: 1,
+    });
+
+    let encrypted_tally = EncryptedVoteTally::new(vote_plan.to_id());
+    let fragment = controller
+        .fragment_factory()
+        .vote_encrypted_tally(&alice, encrypted_tally);
+
+    let parameters = ledger.parameters.clone();
+    let date = ledger.date();
+
+    c.bench_function("vote_encrypted_tally", |b| {
+        b.iter(|| {
+            ledger
+                .ledger
+                .apply_fragment(&parameters, &fragment, date)
```

Comment:
It's not super clear at first sight why this does not save the result for me, so maybe add a comment?